testplan/testing/base.py

```diff
@@ -412,24 +511,115 @@ def start_test_resources(self):
         The base implementation is very simple but may be overridden in sub-
         classes to run additional setup pre- and post-environment start.
         """
-        self.make_runpath_dirs()
-        self.resources.start()
+        # in case this is called more than once from interactive
+        self.report.timer.clear()
+
+        self._add_step(self.setup)
```

Comment:
These steps are nicely listed in this order in the _run_batch_steps, consider to refactor,

same for the stop_test_resources.

This would share more logic between interactive and non interactive

Reply:
but _run_batch_steps goes all the way to test execution, here we just need the steps to start/stop_test_resources.

Reply:
yes but the batch steps is really compsed of start resources then run batch step than stop resources, so _run_batch_steps can reuse these guys, instead of trying to keep them in sync